browser/resources/settings/email_aliases_page/tests/email_aliases.test.tsx

```diff
@@ -0,0 +1,173 @@
+  },
+  formatMessage: (key: string, params: Record<string, string>) => {
+    return key
+  },
+  formatLocale: (key: string, params: Record<string, string>) => {
+    return key
+  }
+}))
+
+// Mock the email aliases service
+class MockEmailAliasesServiceInterface extends EmailAliasesServiceInterface {
+  private observerInit?:
+    (observer: EmailAliasesServiceObserverInterface) => void
+  constructor(observerInit?:
+    (observer: EmailAliasesServiceObserverInterface) => void) {
+    super()
+    this.observerInit = observerInit
+  }
+
+  addObserver(observer: EmailAliasesServiceObserverInterface) {
+    this.observerInit?.(observer)
+  }
+  removeObserver(observer: EmailAliasesServiceObserverInterface) {}
+
+  generateAlias = jest.fn()
+  updateAlias = jest.fn()
+  deleteAlias = jest.fn()
+  requestPrimaryEmailVerification = jest.fn()
+  cancelPrimaryEmailVerification = jest.fn()
+  logout = jest.fn()
+  showSettingsPage = jest.fn()
+}
+
+const createBindObserver =
+  (emailAliasesService: MockEmailAliasesServiceInterface) =>
+  (observer: EmailAliasesServiceObserverInterface) => {
+    emailAliasesService.addObserver(observer)
+    return () => {
+      emailAliasesService.removeObserver(observer)
+    }
+  }
+
+describe('ManagePage', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('shows loading state initially', async () => {
+    const mockEmailAliasesService = new MockEmailAliasesServiceInterface()
+    await act(async () => {
+      render(<ManagePage
+               emailAliasesService={mockEmailAliasesService}
+               bindObserver={createBindObserver(mockEmailAliasesService)}
+      />)
+    })
+    expect(document.querySelector('leo-progressring'))
+      .toBeInTheDocument()
+    expect(screen.getByText(localeRegex(
+      'emailAliasesConnectingToBraveAccount'))).toBeInTheDocument()
+  })
+
+  it('shows sign up form when no email is available', async () => {
+    const mockEmailAliasesService =
+      new MockEmailAliasesServiceInterface((observer) => {
+      observer.onLoggedOut()
+    })
+    await act(async () => {
+      render(<ManagePage
+               emailAliasesService={mockEmailAliasesService}
+               bindObserver={createBindObserver(mockEmailAliasesService)}
+      />)
+    })
+
+    await waitFor(() => {
+      expect(screen.getByText(localeRegex('emailAliasesSignInOrCreateAccount')))
+        .toBeInTheDocument()
+      expect(screen.getByPlaceholderText(localeRegex(
+        'emailAliasesEmailAddressPlaceholder'))).toBeInTheDocument()
+    })
+  })
+
+  it('shows main view when email is available', async () => {
+    const mockEmail = 'test@brave.com'
+    const mockAliases = [
+      {
+        email: 'alias1@brave.com',
+        note: 'Test Alias 1',
+        domains: undefined,
+      }
+    ]
+    const mockEmailAliasesService =
+      new MockEmailAliasesServiceInterface((observer) => {
+      observer.onLoggedIn(mockEmail)
+      observer.onAliasesUpdated(mockAliases)
```

Comment:
we should only be calling one of these for a unit test, not both

Reply:
The reason there are two here is that the first observer ensures we are showing the "logged in" view, which is necessary to show the list of aliases. (Because when we are not yet logged in, we don't show the alias list.) `onAliasUpdated` makes sure we are show the list of mock aliases.

Reply:
I see, ideally I think tests that depend on being logged in would be in their own describe section with `beforeEach`, but we should at least comment here to explain. Also it looks like the indentation is wrong, are these not auto-formatted by presubmit?

Reply:
After the latest revisions this particular code (calling two observer APIs) no longer exists.